### PR TITLE
feat: update logic for associating a submission to the logged in user

### PR DESCRIPTION
**Pull request by dblandin**
Before:
- Submission email address must match logged in user's email address
- Submission user_id must not already be present

After:
- If user_id is present, it must corresponding to the ID of the logged in user
- If user_id is not present, submission must be in a draft state

cc/ @joeyAghion as you were interested in these permission changes.

### CODE REVIEW

**Comment by dblandin** [comment]
@olerichter00 @nickskalkin This should be ready for review. Added some missing test coverage to exercise the new behavior.

File: app/graphql/resolvers/add_user_to_submission_resolver.rb [hunk 1]
```diff
@@ -6,23 +6,25 @@ class AddUserToSubmissionResolver < BaseResolver
...
     end
 
-    #  make sure submission has no user
     if submission.user_id
       raise(GraphQL::ExecutionError, "Submission already has a user")
     end
 
+    if submission.state != Submission::DRAFT
+      raise(GraphQL::ExecutionError, "Submission must be in a draft state to claim")
+    end
```
Comment (olerichter00) [inline app/graphql/resolvers/add_user_to_submission_resolver.rb:20]: 👍

**Review by olerichter00 — APPROVED** [review]
The code looks good 👍 

I'm just curious why the email check isn't needed anymore. 🤔

File: spec/requests/api/graphql/mutations/add_user_to_submission_spec.rb [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        mutation {
+          addUserToSubmission(input: { id: \"#{submission.id}\" }) {
+            clientMutationId
+            consignmentSubmission {
+              internalID
+            }
+          }
+        }
+    GRAPHQL
+
```
Comment (nickskalkin) [inline spec/requests/api/graphql/mutations/add_user_to_submission_spec.rb:57]: **suggestion**: I would add something like `expect(SubmissionService).to receive(:add_user_to_submission).never` to confirm that it actually no-ops

**Review by nickskalkin — APPROVED** [review]
👍

**Comment by dblandin** [comment]
> I'm just curious why the email check isn't needed anymore.

Since removing the Contact Information step, we're no longer collecting an unauthenticated visitor's email address in the submission flow — so the current email check can no longer be satisfied. At least that's my current understanding.

File: app/graphql/resolvers/add_user_to_submission_resolver.rb [hunk 3]
```diff
@@ -6,23 +6,25 @@ class AddUserToSubmissionResolver < BaseResolver
   def run
     check_submission_presence!
```
Comment (joeyAghion) [inline app/graphql/resolvers/add_user_to_submission_resolver.rb:7]: [Suggested a change here](https://artsy.slack.com/archives/C05EQL4R5N0/p1719515049539169?thread_ts=1718979219.264869&cid=C05EQL4R5N0).

**Pull request merged**